src/Uno.UI/UI/Xaml/Controls/TextBlock/TextVisual.skia.cs

```diff
@@ -12,23 +12,27 @@
 using Windows.UI.Text;
 using Windows.UI.Xaml.Controls;
 using Windows.UI.Xaml.Media;
+using Uno.UI.Composition;
 
 #nullable enable
 
 namespace Windows.UI.Composition
 {
 	internal class TextVisual : Visual
 	{
-		private readonly TextBlock _owner;
+		private readonly WeakReference<TextBlock> _owner;
```

Comment:
Shouldn't `TextVisual` and `TextBlock` has the same life-time?

If I understand correctly, the need for `WeakReference` indicates that `TextVisual` lives longer than `TextBlock`, and we want the `TextBlock` to be GC'ed. But how can `TextVisual` live longer than `TextBlock`?

Reply:
It does not really lives longer : one of the parents of the `TextVisual` will be un-referenced so it's going to be collected at some point. But as now the `VisualCollection` properly sets `Visual.Parent` (and clear it on remove) we have 2 circular references: between visual and its parent and between the `TextBlock` and its `TextVisual`. The GC seems to not be able to resolve them.